internal/app/handlers.go

```diff
@@ -460,13 +461,86 @@ func (app *Application) home(w http.ResponseWriter, r *http.Request) {
 	component.Render(r.Context(), w)
 }
 
-func (app *Application) searchResults(w http.ResponseWriter, r *http.Request) {
-	component := components.SearchResults("Search", "Hepatitis")
+func (app *Application) formSearch(w http.ResponseWriter, r *http.Request) {
+	r.ParseForm()
+
+	searchTerm := r.Form["search"][0]
+	searchType := r.Form["options"][0]
+
+	search(w, r, app.repository, searchTerm, searchType, app.logger)
+}
+
+func search(w http.ResponseWriter, r *http.Request, rp *repository.Repository, searchTerm, searchType string, logger *slog.Logger) {
+	var result = &models.CodeSystemResultRow{}
+	defaultPageCount := 5
+
+	// retrieve code system
+	codeSystem, err := rp.GetCodeSystemsByLikeOID(r.Context(), searchTerm)
+	fmt.Println(codeSystem, err)
+	if err != nil || len(*codeSystem) < 1 {
+		if err == nil {
+			err = sql.ErrNoRows
+		}
+		handleError(err, searchTerm, logger, w, r)
+	}
+
+	for _, cs := range *codeSystem {
+		result.CodeSystems = append(result.CodeSystems, &cs)
+	}
+
+	if len(result.CodeSystems) <= defaultPageCount {
+		defaultPageCount = len(result.CodeSystems)
+	}
+
+	result.PageCount = defaultPageCount
+	result.CodeSystemsCount = strconv.Itoa(len(result.CodeSystems))
+
+	// // retrieve concepts that are part of that code system
+	// concepts, err := rp.GetCodeSystemConceptsByCodeSystemOID(r.Context(), app.db, codeSystem)
+	// for _, csc := range *concepts {
+	// 	result.CodeSystems = append(result.CodeSystems, &csc)
+	// }
+	// result.CodeSystemConcepts = concepts
+
+	// for now
+	result.CodeSystemConceptsCount = strconv.Itoa(0)
+
+	// for now
+	result.ValueSetsCount = strconv.Itoa(0)
+
+	w.Header().Set("HX-Push-Url", fmt.Sprintf("/search?type=%s&input=%s", searchType, searchTerm))
+
+	component := components.SearchResults(true, "Search", searchTerm, result)
 	component.Render(r.Context(), w)
 }
 
+func (app *Application) directSearch(w http.ResponseWriter, r *http.Request) {
+	searchType := r.URL.Query().Get("type")
+	searchTerm := r.URL.Query().Get("input")
+	search(w, r, app.repository, searchTerm, searchType, app.logger)
+}
+
 func (app *Application) handleBannerToggle(w http.ResponseWriter, r *http.Request) {
 	action := r.PathValue("action")
 	component := components.UsaBanner(action)
 	component.Render(r.Context(), w)
 }
+
+func handleError(err error, input string, logger *slog.Logger, w http.ResponseWriter, r *http.Request) {
+	if errors.Is(err, sql.ErrNoRows) {
+		errorString := fmt.Sprintf("Error: Code System %s not found", input)
+		dbErr := &customErrors.DatabaseError{
+			Err:    err,
+			Msg:    errorString,
+			Method: "getCodeSystemById",
+			Id:     input,
+		}
+		component := components.Error("Search", dbErr.Msg)
+		component.Render(r.Context(), w)
+		dbErr.NoRows(w, r, err, logger)
```

Comment:
Looks like the second `Error: Code System 1.0.1 not found` at the bottom of the page is coming from this line.

Reply:
@jakewheeler updated!